client/blocks/inline-help/popover.jsx

```diff
@@ -32,46 +35,66 @@ class InlineHelpPopover extends Component {
 	};
 
 	state = {
-		showContactForm: false,
+		showSecondaryView: false,
+		activeSecondaryView: '',
 	};
 
-	openResult = ( event, href ) => {
-		if ( ! href ) {
-			return;
-		}
-
-		this.props.recordTracksEvent( 'calypso_inlinehelp_link_open', {
-			search_query: this.props.searchQuery,
-			result_url: href,
-		} );
-
-		if ( ! event.metaKey ) {
-			event.preventDefault();
-			window.location = href;
-		} else if ( event.key === 'Enter' ) {
-			event.preventDefault();
-			window.open( href, '_blank' );
-		}
+	openResultView = event => {
+		event.preventDefault();
+		this.openSecondaryView( VIEW_RICH_RESULT );
 	};
 
 	moreHelpClicked = () => {
 		this.props.onClose();
 		this.props.recordTracksEvent( 'calypso_inlinehelp_morehelp_click' );
 	};
 
-	toggleContactForm = () => {
-		if ( this.state.showContactForm ) {
-			this.props.recordTracksEvent( 'calypso_inlinehelp_contact_hide' );
-		} else {
-			this.props.recordTracksEvent( 'calypso_inlinehelp_contact_show' );
-		}
-		this.setState( { showContactForm: ! this.state.showContactForm } );
+	setSecondaryViewKey = secondaryViewKey => {
+		this.setState( { activeSecondaryView: secondaryViewKey } );
+	};
+
+	openSecondaryView = secondaryViewKey => {
+		this.setSecondaryViewKey( secondaryViewKey );
+		this.props.recordTracksEvent( `calypso_inlinehelp_${ secondaryViewKey }_show` );
+		this.setState( { showSecondaryView: true } );
+	};
+
+	closeSecondaryView = secondaryViewKey => {
+		this.setSecondaryViewKey( '' );
+		this.props.recordTracksEvent( `calypso_inlinehelp_${ secondaryViewKey }_hide` );
+		this.props.selectResult( -1 );
+		this.setState( { showSecondaryView: false } );
+	};
+
+	openContactView = () => {
+		this.openSecondaryView( VIEW_CONTACT );
+	};
+
+	handleCloseSecondaryViewButton = () => {
+		this.closeSecondaryView( this.state.activeSecondaryView );
+	};
+
+	renderSecondaryView = () => {
+		const classes = classNames(
+			'inline-help__secondary-view',
+			`inline-help__${ this.state.activeSecondaryView }`
+		);
+		return (
+			<div className={ classes }>
+				{
+					{
+						contact: <HelpContact compact selectedSite={ this.props.selectedSite } />,
+						richresult: <InlineHelpRichResult result={ this.props.selectedResult } />,
+					}[ this.state.activeSecondaryView ]
```

Comment:
Though I do like this pattern, I can see this pattern getting out of hand in other situations (bigger &/ more components).
In this case it's readable, but that there is 'logic' albeit only a small amount, suggests to me that this would ideally be another component.

Not a blocker, just thinking out loud again and offering thoughts on the usage of this pattern.